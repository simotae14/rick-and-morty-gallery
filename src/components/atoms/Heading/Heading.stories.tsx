import type { Meta, StoryObj } from '@storybook/react';

import Heading from './index';

const meta = {
  title: 'Atoms/Heading',
  component: Heading,
  tags: ['autodocs'],
  argTypes: {
    title: {
      control: {
        type: 'text',
      },
    },
  },
} satisfies Meta<typeof Heading>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Base: Story = {
  args: {
    title: 'Rick and Morty Gallery',
  },
};
