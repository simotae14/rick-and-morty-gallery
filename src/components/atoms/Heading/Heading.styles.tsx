import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const LinkWrapper = styled(Link)`
  text-decoration: none;
  color: inherit;
`;

export const Title = styled.h1`
  font-weight: 900;
  font-size: 3rem;
  padding: 32px 16px;
  margin: 0;
  max-width: 1220px;
  @media (max-width: 1220px) {
    max-width: 880px;
    margin: auto;
  }
  @media (max-width: 912px) {
    max-width: 576px;
    margin: auto;
  }
`;
