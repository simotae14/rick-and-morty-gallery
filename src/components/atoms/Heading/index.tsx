import { Title } from './Heading.styles';

interface HeadingProps {
  /**
   * Is this the possible title of the page
   */
  title: string;
}

/**
 * Component used for the Heading of the page
 */
function Heading({ title }: HeadingProps) {
  return <Title>{title}</Title>;
}

export default Heading;
