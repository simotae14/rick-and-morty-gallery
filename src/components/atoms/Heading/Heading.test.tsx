/* eslint-disable import/no-extraneous-dependencies */
import { describe, it } from 'vitest';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { axe, toHaveNoViolations } from 'jest-axe';

import Heading from '.';

expect.extend(toHaveNoViolations);

describe('Components => atoms => Heading', () => {
  it('should render the Heading', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/']}>
        <Heading title="title" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not fail any accessibility tests', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <Heading title="title" />
      </MemoryRouter>
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});
