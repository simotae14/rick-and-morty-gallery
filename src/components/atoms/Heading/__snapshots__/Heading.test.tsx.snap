// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components => atoms => Heading > should render the Heading 1`] = `
<DocumentFragment>
  <a
    class="sc-eLqIir cirJmn"
    href="/"
  >
    <h1
      class="sc-HzRMQ bubYoD"
    >
      title
    </h1>
  </a>
</DocumentFragment>
`;
